tools/plugins/com.liferay.ide.project.core/src/com/liferay/ide/project/core/PluginsSDKProjectProvider.java

```diff
@@ -208,6 +212,8 @@ public IStatus doCreateNewProject( NewLiferayPluginProjectOp op, IProgressMonito
         // need to update project name incase the suffix was not correct
         op.setFinalProjectName( newProject.getName() );
 
+        projectNames.insert().setName( op.getFinalProjectName().content() );
```

Comment:
Good job this is the missing piece, now both maven and SDK providers use the new projectNames and also the value added to finalProjectName will get correctly added here.